correlation/src/main/java/com/cerner/beadledom/correlation/CorrelationContextImpl.java

```diff
@@ -0,0 +1,32 @@
+package com.cerner.beadledom.correlation;
+
+import java.util.UUID;
+import javax.annotation.Nullable;
+
+/**
+ * Default implementation that stores the correlation context in {@link ThreadLocal}.
+ */
+public class CorrelationContextImpl implements CorrelationContext {
```

Comment:
Do you think we need a separate interface and default implementation at this time? Would it be worth deferring that decision by making `CorrelationContext` be a class marked `final` so it's the only implementation, but can't be extended leaving us room to split it later if there is demand?

Reply:
This might also require a private constructor with a factory method for creation as well since splitting later would result in moving the constructor to the implementation.

Reply:
I can do that. I was thinking that people may want to eventually use https://github.com/open-telemetry/opentelemetry-java/tree/master/api/src/main/java/io/opentelemetry/correlationcontext

Reply:
Yeah, I could see that. Another option instead of what I suggested above could be to name this `ThreadLocalCorrelationContext`.

Reply:
Either way I think we can implement this library in such a way that we are open to implementing some of this with opentelemetry in the future as a passive change.

Reply:
Renamed to `ThreadLocalCorrelationContext` c1774fd5

Reply:
Switched to be a final class CorrelationContext, c1f9800a